packages/cloud_firestore_odm/cloud_firestore_odm_generator/lib/src/collection_data.dart

```diff
@@ -157,41 +159,11 @@ class CollectionData with Names {
       );
     }
 
-    final annotatedElementSource = annotatedElement.librarySource;
-
-    // TODO(rrousselGit) handle parts
-    // Whether the model class and the reference variable are defined in the same file
-    // This is important because json_serializable generates private code for
-    // decoding a Model class.
-    final modelAndReferenceInTheSameLibrary =
-        collectionTargetElement.librarySource == annotatedElementSource;
-
-    final fromJson = collectionTargetElement.constructors.firstWhereOrNull(
-      (ctor) => ctor.name == 'fromJson',
-    );
-    if (hasJsonSerializable && !modelAndReferenceInTheSameLibrary) {
```

Comment:
Why delete all of this?

Reply:
Since we require every class to have JsonSerializable on it I was trying to clean this up. At least some of this should probably stay though.